### PR TITLE
[T2A5][W13-A1]Yong Zhi Yuan

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Not easy doing this :(

### CODE REVIEW

File: test/expected.txt [hunk 1]
```diff
@@ -139,8 +139,9 @@
 || 0 persons found!
 || ===================================================
 || Enter command: || [Command entered:  find betsy]
```
Comment (okkhoy) [inline test/expected.txt:144]: errr... update expected.txt :-D not "updated"

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -358,7 +373,30 @@ public static String executeCommand(String userInputString) {
         }
     }
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:382]: going forward, please note that you need to add header comments when you add a new method.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -373,11 +380,71 @@ public static String executeCommand(String userInputString) {
...
+     * @return feedback display message for the operation result
+     */
     private static String executeEdit(String commandArgs) {
-    	final Optional<String[]> decodeResult = decodePersonFromString(commandArgs);
-    	final String[] personToAdd = decodeResult.get();
+        String message = checkIfEditUserExists(commandArgs);
+        if(message.equals(MESSAGE_INPUT_NEW_DETAILS)){
+            String newDetails = inputNewDetails(message);
+            return addEditPerson(newDetails);
+        }
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:394]: ta! daaaa! coding standard violation ;-) 
need to have the `else` "cuddled" between the braces `} else {`

**Comment by okkhoy** [comment]
@Zhiyuan-Amos good job with this PR :+1: 
- i noticed some coding standard violations (pl. see code comments)

some food for thought:
- how would using a hash map make this code different? (easy or difficult?) you could have carried over T1 activity here as well instead of using string arrays
- what else can you do to enhance this address book (apart from edit, sort, find?) just think and comment here if time permits, no need to code all that :-)

File: src/seedu/addressbook/AddressBook.java (hunk 3 above)
Comment (Zhiyuan-Amos) [reply src/seedu/addressbook/AddressBook.java:394]: Ah okay noted! Thanks much for your review! :)

**Comment by Zhiyuan-Amos** [comment]
Thanks for your review Akshay! :D 

Will converting it to hashmap make the code better? The implementation seems pretty the same to me, since person[0], [1], [2] refers to the name, phone and email, while person.get("name"), person.get("phone"), person.get("email") returns the same results as well, and this is implemented using ArrayList<HashMap>. Am I missing something? ><

Possible ways to enhance addressbook: 
1. sort can be done in alphabetical order for email addresses
2. mass import the details (name, email address and phone number) from Yahoo email etc? =0
3. allow natural language + variations: "add" / "a" / "insert" etc refers to the same command

**Comment by okkhoy** [comment]
The advantage of doing it via HashMap is code readability; other than that
there is not much that makes it better;
This stackoverflow answer
http://stackoverflow.com/questions/6462055/hashmap-vs-array-performancewill
give an idea (you can dig deeper if you wish ;-) )
​
Among the enhancements I like the idea of 1  and 3; 2 well, it depends what
is the use-case!

**Comment by Zhiyuan-Amos** [comment]
Noted, thanks again Akshay! :)

See you in class tomorrow!

**Pull request closed without merging**